include/ofi_util.h

```diff
@@ -1286,6 +1286,7 @@ struct ofi_ops_flow_ctrl {
 };
 
 struct util_rx_entry {
+	struct dlist_entry	entry;
```

Comment:
If you make this:
```
union {
    struct dlist_entry dentry;
    struct slist_entry sentry;
 }
```
I think it'll work in the debug case without abusing the fields.

Reply:
Yeah I considered that and it'll help clean up the code to remove the casts but I don't see why that will have a functional difference?

Reply:
There's 'functional' difference if you print the structure in a debugger in terms of being able to read the output when the structure is on an slist.  The behavior in the code at runtime is the same.  In terms of code readability, this makes it clear that dlist and slist entries overlap and the larger struct may be stored on both.